# RNAseq_Analysis
*Basic RNAseq analysis workflow*

## A - PREPROCESSING 
##### *Structure*

```py
create_repository.sh
```
```
project_name/
ğŸ“ 1.raw_data/             â€“ raw data (.fq.gz)
   ğŸ“ md5sum/        â€“ md5sum files check and copied
   ğŸ“ trimmed_data/         â€“ fastq after trimming -optionnal (.fq.gz)
ğŸ“ 2.fastqc_results/       â€“ fastQC results (html)
   ğŸ“ trimmed/           â€“ FastQC sur fichiers trimmed  
ğŸ“ 3.star/
   ğŸ“ bam/               â€“ fichiers log                 
ğŸ“ 4.featurecounts/        â€“ featurecounts results (txt)  
ğŸ“ logs/                 â€“ fichiers log
ğŸ“ scripts/              â€“ scripts bash 
```
##### *Md5sum*
```py
md5sum_check.sh
```
- creates md5sum of the copied fastq
- check if the md5sum are matching in `md5sum.out`
- writes in `md5sum.out` the files that are not matching

##### *Quality check*
```py
qualitycheck_fastqc.sh
```
- run fastqc for all raw data (*fq.gz)
- creates html report for each fastq with
For more informations about fastqc check https://www.bioinformatics.babraham.ac.uk/projects/fastqc/

##### *Trimming (optionnal)*

## B - PROCESSING 
##### *Alignment - STAR*
1 - Build STAR
- gtf file needed
```py
build_STAR_index.sh
```
2 - Proceed to the alignment
```py
STAR_alignment.sh
```
##### *Counting features - featurecounts*
1 - Sort bam

```py
sort_bam.sh
```
```py
featurecounts.sh
```
## C - R analysis with Deseq2
- Exploring Dataset
- Create Deseq object
- Visualization
- DEgs identification
- Pathways identification
