# RNAseq_Analysis
*Basic RNAseq analysis workflow*

## A - PREPROCESSING 
##### *Structure*

```py
create_repository.sh
```
```
project_name/  
ğŸ“ raw_data/             â€“ fichiers bruts (.fastq.gz)  
ğŸ“ trimmed_data/         â€“ fichiers aprÃ¨s trimming  
ğŸ“ fastqc_results/       â€“ rÃ©sultats FastQC  
   ğŸ“ raw/               â€“ FastQC sur fichiers bruts  
       ğŸ“ md5sum/        â€“ md5sum files check and copied  
   ğŸ“ trimmed/           â€“ FastQC sur fichiers trimmed  
ğŸ“ multiqc_results/      â€“ rÃ©sultats MultiQC  
ğŸ“ logs/                 â€“ fichiers log  
ğŸ“ scripts/              â€“ scripts bash / Python
```
##### *Md5sum*
```py
md5sum_check.sh
```
- creates md5sum of the copied fastq
- check if the md5sum are matching in `md5sum.out`
- writes in `md5sum.out` the files that are not matching

##### *Quality check*
```py
qualitycheck_fastqc.sh
```
- run fastqc for all raw data (*fq.gz)
- creates html report for each fastq with
For more informations about fastqc check https://www.bioinformatics.babraham.ac.uk/projects/fastqc/

##### *Trimming (optionnal)*

## B - PROCESSING 
##### *Alignment - STAR*
1 - Build STAR
- gtf file needed
```py
build_STAR_index.sh
```
2 - Proceed to the alignment
```py
STAR_alignment.sh
```
##### *Counting features - HTSeqcount*

## C - Statistical analysis 
