# RNAseq_Analysis
*Basic RNAseq analysis workflow*


## A - PREPROCESSING 
```py
create_repository.sh
```
### *File organisation for FastQC analysis*
project_name/  
ğŸ“ raw_data/             â€“ fichiers bruts (.fastq.gz)  
ğŸ“ trimmed_data/         â€“ fichiers aprÃ¨s trimming  
ğŸ“ fastqc_results/       â€“ rÃ©sultats FastQC  
   ğŸ“ raw/               â€“ FastQC sur fichiers bruts  
   ğŸ“ trimmed/           â€“ FastQC sur fichiers trimmed  
ğŸ“ multiqc_results/      â€“ rÃ©sultats MultiQC  
ğŸ“ logs/                 â€“ fichiers log  
ğŸ“ scripts/              â€“ scripts bash / Python  

###1 - Receive the fastq from a sequencing commpagny
Create a directory containing all the fastq and compare md5sum with bash code

###2 - Quality check #################################
Controle the quality sequencing with fastqc

###3 - Trimming (optionnal)

B - PROCESSING #################################
###1 - Alignment with STAR

###2 - Counting features with HTSeqcount

C - STATISTICAL ANALYSIS #################################
